### DISCUSSION TITLE
Prometheus metrics endpoint for monitoring purposes

### DISCUSSION THREAD

**Discussion by badihi** [postalserver/postal#1708] [Feature suggestions]
I assume a metrics exporter is fundamental to such a platform.
As it requires a broad understanding of the system and involves the whole system, I think it must be done by the main authors.

**Comment by willpower232** [comment]
There is already integration with Graylog (https://www.graylog.org/) which may solve some monitoring concerns.

You could also configure Prometheus on your server to monitor Docker directly (https://docs.docker.com/config/daemon/prometheus/) which may be more useful than integrating with Prometheus in the code itself.

**Comment by badihi** [comment]
I really doubt that docker monitoring is useful at least in my case.
I'm experiencing regular problems such as sending emails being stopped that is resolved by a restart and I could not find the reason yet. 
These kind of problems can be monitored by having metrics like in/out rate and success rate of email sendings. Obviousely such metrics cannot be obtained using docker monitoring or any other general monitoring solutions.
I'll take a look at graylog integration and will let you know if it was helpful.

**Comment by badihi** [comment]
Also we can consider developing a simple metric exporter sidecar. It will be possible if apis to get these metrics are available. Otherwise the only way is to directly connect to the database and query metrics from there.
It would be appreciated if you anyone can share resources or docs (if any) about mentioned apis here. It will be a good starting point.
